js_methods.rb

```diff
@@ -0,0 +1,253 @@
+# frozen_string_literal: true
+
+# rubocop:disable Style/StringLiterals, Style/AndOr, CyclomaticComplexity, PerceivedComplexity, ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    i = 0
+    while i < length
+      yield to_a[i] if block_given?
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    i = 0
+    while i < length
+      yield to_a[i], i if block_given?
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    temp_arr = []
+    my_each do |e|
+      temp = yield e
+      temp_arr << e if temp
+    end
+    temp_arr
+  end
+
+  def my_all?(*vars)
+    ans = true
+    if block_given?
+      my_each { |e| ans = false unless yield e }
+    elsif !vars.empty?
+      if vars[0].is_a?(Class) # 2/4
+        my_each { |e| ans = false unless e.is_a?(vars[0]) }
+      elsif vars[0].is_a?(Regexp) # 3/4
+        my_each { |e| ans = false unless e =~ vars[0] }
+      else
+        my_each { |e| ans = false unless e == vars[0] } # 4/4
+      end
+    else
+      my_each { |e| ans = false if e == false || e.nil? } # 1/4
+    end
+    ans
+  end
+
+  def my_any?(*vars)
+    !my_none?(*vars)
+  end
+
+  def my_none?(*vars)
+    ans = true
+    if block_given?
+      my_each { |e| ans = false if yield e }
+    elsif !vars.empty?
+      if vars[0].is_a?(Class) # 2/4
+        my_each { |e| ans = false if e.is_a?(vars[0]) }
+      elsif vars[0].is_a?(Regexp) # 3/4
+        my_each { |e| ans = false if e =~ vars[0] }
+      else
+        my_each { |e| ans = false if e == vars[0] } # 4/4
+      end
+    else
+      my_each { |e| ans = false if e != false || e.nil? } # 1/4
+    end
+    ans
+  end
```

Comment:
Same issue here, since they are linked. There seem not to be a branch to handle for when a `proc` is passed.

Are you sure `line 72` is doing what it should? `[false, nil].my_none?` should return `true`.

Reply:
Done. The proc branch is not needed. The code works fine with a proc as it is. Anyways fixed other issues with the my_none. Thanks.